### ISSUE TITLE
2020 [NIPS] Swapping Autoencoder for Deep Image Manipulation

### ISSUE THREAD

**Issue by pomelyu** [pomelyu/paper-reading-notes#16]
## Introduction
<img width="1109" alt="image" src="https://github.com/pomelyu/paper-reading-notes/assets/2679561/6409b1f7-1c52-461b-aa33-de5b5fd0d608">

https://github.com/taesungp/swapping-autoencoder-pytorch
This paper proposed an auto-encoder structure and an unsupervised training mechanism to disentangle the **structure** and **style** components of an image. Based on the disentanglement, new realistic images can be generated by smoothly switching between different styles or modifying the structure content. Unlike using the unconditional generative networks, such as StyleGAN2, the completely feed-forward structure has lower inference time and preserves the original content better.

## Method
<img width="1006" alt="image" src="https://github.com/pomelyu/paper-reading-notes/assets/2679561/d667abbf-dc76-4a2c-9a05-2f0c5c97312e">
<img width="1267" alt="image" src="https://github.com/pomelyu/paper-reading-notes/assets/2679561/81e620da-ab74-438d-9735-4aaaf38d8a12">

- Generator, Discriminator from StyleGAN2
- Encoder encodes the input into **structure** and **style** components. 
   - **structure** component is a 3D(DxHxW) vector that records the spatial content. This branch contains only convolution block to let the information encoded locally
   - **style** component is a 1D vector that records the global information. This branch uses only valid or reflective padding and applies average pooling in the end of the structure to ignore the unnecessary local information
- The discriminator forces the reconstruction images and the style-swapped image look realistic
- Since the style is a global information, we can assume every patch in the image shares this same information. The co-occurrence discriminator takes patches from the style-swapped image and forces them look similar to the patches of the desired style image.

## Highlight
- Vector arithmetic editing
- Unsupervised training, no labels required
- Doesn't make any assumption to the latent space distribution but still gets good disentangle
- Be able to efficiently edit the existing images thanks to the feed forward method
- The edition can be applied on only part of the image thanks to the structure component.
- Author provided the comparison between patch size of the co-occurrence discriminator and the final outcomes.

## Limitation
- The definition of "style" is quite ambiguous. Further, as other image-guided editing method, it is difficult to perform the precise editing. For example, Which one are glasses belong to, style or structure? According to the examples in the paper, glasses belong to style. At little weird 

## Comments
- Not sure how to perform the gaze or mouth manipulation. Perform warping on the 32x32 latent space might not be practical. Also how to perform light editing
<img width="1085" alt="image" src="https://github.com/pomelyu/paper-reading-notes/assets/2679561/019a920a-d6f0-436b-a3ba-af8e8696ea8b">
